### PR TITLE
Fix Markdown rendering

### LINKED ISSUE #38: Regression: Wrong type during markdown rendering
After https://github.com/hashicorp/terraform-config-inspect/commit/c6ae6269b9d76f3d7b4bd6319d9879cb5f15f2ec landed, rendering markdown now appears to fail with the following:

```
error rendering template: template: md:15:47: executing "md" at <$versions>: wrong type for value; expected []string; got *tfconfig.ProviderRequirement
```

### PR DESCRIPTION
**Pull request by radeksimko**
Fixes #38 

---

I took the liberty to decouple the markdown rendering logic and added Markdown test data for all relevant test cases.

### CODE REVIEW

**Comment by mildwonkey** [comment]
thanks @radek! This looks good. I have a couple of questions but they aren't blockers:

1: What was the markdown bug you fixed? The branch and commit message indicate a bug.
2: Did you mean to omit a test out.md for `basics-json`?

**Comment by radeksimko** [comment]
1. The bug is described in #38 - basically most (all?) Markdown rendering logic was broken, probably a side effect of that logic being entirely untested. 🤷‍♂ 
2. Kind of. That test seems to be testing the parser (whether it decodes JSON and HCL equally well), but not the actual rendering logic - be it JSON or Markdown. So I assume the existing test already covers that and additional Markdown test would bring little value (if it ever breaks I suppose we'd see it in the JSON test). I'm happy to add it though, if you think it's safer.

File: tfconfig/markdown.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Core Version Constraints:
+{{- range .RequiredCore }}
+* {{ tt . }}
+{{- end}}{{end}}
+
+{{- if .RequiredProviders}}
+
+Provider Requirements:
+{{- range $name, $req := .RequiredProviders }}
+* **{{ $name }}{{ if $req.Source }} ({{ $req.Source | tt }}){{ end }}:** {{ if $req.VersionConstraints }}{{ commas $req.VersionConstraints | tt }}{{ else }}(any version){{ end }}
```
Comment (radeksimko) [inline tfconfig/markdown.go]: @mildwonkey These two lines are the only difference from the previous template.

**Comment by mildwonkey** [comment]
🤔 Thanks for pointing out the change - can you add a test example with a provider source attribute, and (if you haven't already) make sure the registry folks know that output is changing? I believe they parse that field.

**Comment by mildwonkey** [comment]
also I am so sorry, you linked the issue right in this PR 🙄 I need to invite _all_ my braincells to monday morning PR reviews next time

**Comment by radeksimko** [comment]
> can you add a test example with a provider source attribute

I think this test covers that already?
https://github.com/hashicorp/terraform-config-inspect/pull/39/files#diff-76fee4570d37301e18d103509279bacfR5

**Comment by radeksimko** [comment]
> make sure the registry folks know that output is changing

I requested review from @hashicorp/tf-self-serve - I'm not sure if we have any better team on GitHub covering the Registry.

FWIW the output is not changing for providers that do not have a source defined. It's rather the internal representation which broke markdown rendering.

**Pull request merged**